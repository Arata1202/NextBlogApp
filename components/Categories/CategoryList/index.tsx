import { Tag } from '@/libs/microcms';
import CategoryListItem from '../CategoryListItem';
import styles from './index.module.css';

type Props = {
  tags?: Tag[];
  hasLink?: boolean;
};

export default function CategoryList({ tags, hasLink = true }: Props) {
  if (!tags) {
    return null;
  }
  return (
    <ul className={styles.tags}>
      {tags.map((tag) => (
        <li key={tag.id}>
          <CategoryListItem tag={tag} hasLink={hasLink} />
        </li>
      ))}
    </ul>
  );
}
