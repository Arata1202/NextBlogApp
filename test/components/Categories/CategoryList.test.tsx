import { render } from '@testing-library/react';
import CategoryList from '@/components/Categories/CategoryList';

describe('CategoryList', () => {
  test('CategoryList)', () => {
    const { asFragment } = render(<CategoryList />);
    expect(asFragment()).toMatchSnapshot();
  });
});
