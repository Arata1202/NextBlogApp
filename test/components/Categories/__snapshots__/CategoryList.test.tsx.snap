// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryList CategoryList) 1`] = `<DocumentFragment />`;
